cmd/contour/serve.go

```diff
@@ -303,9 +315,25 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	eh.Metrics = metrics
 	eh.CacheHandler.Metrics = metrics
 
+	synced := make([]cache.InformerSynced, 0, len(informers))
+	for _, inf := range informers {
+		synced = append(synced, inf.HasSynced)
+	}
+
 	// step 13. create grpc handler and register with workgroup.
 	g.Add(func(stop <-chan struct{}) error {
 		log := log.WithField("context", "grpc")
+
+		log.Printf("waiting for informer caches")
+		cache.WaitForCacheSync(stop, synced...)
+		log.Printf("informer caches synced")
+
+		// Wait for cache to process
+		timeout := 15 * time.Second // Default timeout if reached code still progresses
+		if !eh.WaitForCacheProcess(timeout) {
+			log.Printf("!! timed out waiting %v for cache to process !!", timeout)
```

Comment:
what is the goal of this line? is this a warning? should contour wait longer if the timeout is reached?

Reply:
I didn't want the wait for the processing to hold forever and block contour. My thought was to have a reasonable high default (this case 15 secs). If that time is reached then move along and start the gRPC server.

Reply:
But then the server starts and the cache isn't synced. Ideally we don't want to use a timeout, we want to issue a single operation then wait for it to complete.